File: Web4/webpages/src/main/js/com/fitbank/webpages/behaviors/HTMLPopup.js
include("lib.prototype");
include("fitbank.ui.ventana");
include("fitbank.util");

/**
 * @remote
 */
HTMLPopup.addMethods({

    initialize: function(parametros) {
        copyConstructor.bind(this)(parametros);

        c && c.$N(this.elementName).each(this.init, this);
    },

    init: function(elemento) {
        var iframeHeight = this.height;
        var iframeWidth = this.width;

        if (!this.height || !this.width) {
            iframeHeight = 200;
            iframeWidth = 200;
        }

        var iframe = new Element("iframe" ,{
            name : "HTMLIframe",
            height: iframeHeight + "px",
            width: iframeWidth + "px",
            style: "background-color: white; border: 1px"
        });

        var boton = new Element("input" , {
            type: "button",
            value: this.label
        });

        elemento.hide().insert({
            after: boton
        });

        boton.on('click', (function() {  
            var ventana = new Ventana({
                titulo: "Contenido HTML",
                contenido: iframe
            });

            Util.getContentWindow(iframe).document.body.innerHTML = "";
            Util.getContentWindow(iframe).document.write(elemento.value);

            Element.fireDOMEvent.defer(elemento, "click");   

            ventana.ver();
        }).bind(this));
    }

});

File: Web4/webpages/src/main/java/com/fitbank/webpages/behaviors/HTMLPopup.java
package com.fitbank.webpages.behaviors;

import com.fitbank.webpages.AbstractJSBehaivor;
import com.fitbank.util.Editable;

/**
 * Popup en html
 *
 * @author FitBank JB
 */
public class HTMLPopup extends AbstractJSBehaivor {
    
    @Editable
    private String label = "";

    @Editable
    private String height = "";

    @Editable
    private String width = "";

    public void setLabel(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }

    public String getHeight() {
        return height;
    }

    public void setHeight(String height) {
        this.height = height;
    }

    public String getWidth() {
        return width;
    }

    public void setWidth(String width) {
        this.width = width;
    }

}
